src/h5web/visualizations/nexus/utils.ts

```diff
@@ -68,7 +83,43 @@ export function findNxDataGroup(
   return findNxDataGroup(defaultEntity, metadata);
 }
 
-export function findSignalName(group: HDF5Group): string {
+export function findMyNxDataGroup(group: MyHDF5Group): MyHDF5Group | undefined {
+  if (getAttributeValue(group, 'NX_class') === 'NXdata') {
+    return group; // `NXdata` group found
+  }
+
+  const defaultPath = getAttributeValue(group, 'default');
+  if (defaultPath === undefined) {
+    return undefined; // group has no `default` attribute
+  }
+
+  assertStr(defaultPath, `Expected 'default' attribute to be a string`);
+
+  const isAbsolutePath = defaultPath.startsWith('/');
+  const pathSegments = defaultPath.slice(isAbsolutePath ? 1 : 0).split('/');
+
+  const defaultEntity = pathSegments.reduce<MyHDF5Entity | undefined>(
+    (parentEntity, currSegment) => {
+      return parentEntity && isMyGroup(parentEntity)
+        ? getChildEntity(parentEntity, currSegment)
+        : undefined;
+    },
+    isAbsolutePath ? group.parents[0] : group
+  );
+
+  assertDefined(defaultEntity, `Expected entity at path "${defaultPath}"`);
+  assertMyGroup(defaultEntity, `Expected group at path "${defaultPath}"`);
+
+  const nxDataGroup = findMyNxDataGroup(defaultEntity);
+  assertDefined(
+    nxDataGroup,
+    `Expected NXdata group or group with 'default' attribute`
+  );
```

Comment:
I've added this assertion because I realised that it is still needed in the case where a `default` attribute points to an entity that is neither an NXdata group nor has a `default` attribute.